File: 3612-process-string-with-special-operations-i/solution.cpp
#include <string>
#include <algorithm>
#include <vector>

class Solution {
public:
    std::string processString(std::string s) {
        std::string result = "";
        
        for (char c : s) {
            if (c == '*') {
                if (!result.empty()) {
                    result.pop_back();
                }
            } else if (c == '#') {
                result += result;
            } else if (c == '%') {
                std::reverse(result.begin(), result.end());
            } else {
                // It's a lowercase letter
                result.push_back(c);
            }
        }
        
        return result;
    }
};

File: 3612-process-string-with-special-operations-i/solution.py
class Solution:
    def processString(self, s: str) -> str:
        result = []
        
        for char in s:
            if char == '*':
                if result:
                    result.pop()
            elif char == '#':
                # Duplicate the current result
                result.extend(result)
            elif char == '%':
                # Reverse the current result
                result.reverse()
            else:
                # Append lowercase letter
                result.append(char)
        
        return "".join(result)